src/creatwth.cpp

```diff
@@ -501,35 +510,53 @@ static BOOL UpdateFrom32To64(HANDLE hProcess, HMODULE hModule, WORD machine,
 }
 #endif // DETOURS_64BIT
 
-typedef BOOL(WINAPI *LPFN_ISWOW64PROCESS)(HANDLE, PBOOL);
-
-static BOOL IsWow64ProcessHelper(HANDLE hProcess,
-                                 PBOOL Wow64Process)
-{
-#ifdef _X86_
-    if (Wow64Process == NULL) {
-        return FALSE;
-    }
-
-    // IsWow64Process is not available on all supported versions of Windows.
-    //
-    HMODULE hKernel32 = LoadLibraryW(L"KERNEL32.DLL");
-    if (hKernel32 == NULL) {
-        DETOUR_TRACE(("LoadLibraryW failed: %lx\n", GetLastError()));
-        return FALSE;
-    }
+#ifndef PROCESSOR_ARCHITECTURE_AMD64
+#define PROCESSOR_ARCHITECTURE_AMD64            9
+#endif
 
-    LPFN_ISWOW64PROCESS pfnIsWow64Process = (LPFN_ISWOW64PROCESS)GetProcAddress(
-        hKernel32, "IsWow64Process");
+#ifndef PROCESSOR_ARCHITECTURE_ARM64
+#define PROCESSOR_ARCHITECTURE_ARM64            12
+#endif
 
-    if (pfnIsWow64Process == NULL) {
-        DETOUR_TRACE(("GetProcAddress failed: %lx\n", GetLastError()));
-        return FALSE;
+static BOOL Is64BitOS()
+{
+    BOOL bRet = FALSE;
+    HMODULE hModule = GetModuleHandle(TEXT("kernel32.dll"));
```

Comment:
Suggestion: directly call GetModuleHandleW here

Reply:
No effect, because `TEXT` is the `winnt.h` header file from the Windows SDK, not the` tchar.h` of the CRT

Reply:
And Detours' makefile never defines UNICODE so this goes through WinAPI's internal code page translation when we could just hardcode the wide string instead.

Reply:
Examples in the existing code:
https://github.com/microsoft/Detours/blob/8cbb9e2b822d96d534400da915a5dc258ab65a9a/src/modules.cpp#L58
https://github.com/microsoft/Detours/blob/8cbb9e2b822d96d534400da915a5dc258ab65a9a/src/modules.cpp#L343

Reply:
Okay, follow your Suggestion.